### PR TITLE
V2:astro-942 - Input/Search KM Style Update

### PR DESCRIPTION
**Pull request by im-cr**
**Tickets**
  - https://rocketcom.atlassian.net/browse/ASTRO-942

**Changes** - commit: https://github.com/RocketCommunicationsInc/astro-components/commit/44f88d9c2b736b4523366857ac1204e84184c69d
  - input.css
  - theme--dark
  - theme--light

**Designs**
  - [Dark](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/e4eee8e2-20a2-4977-a7b1-c2d4e6af22ad/files/1d17329d-e1a3-490a-a6ad-097e2ecd4825?filter=changed&s=input)
  - [Light](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/e4eee8e2-20a2-4977-a7b1-c2d4e6af22ad/files/b7c2bf0c-9dec-4b07-96d4-ead85393c9d0?filter=changed&s=input)

### CODE REVIEW

**Comment by amalyah** [comment]
This appears to be using the wrong search icon for search inputs. Do you have access to the other search icon file?

File: src/css/common/_theme--dark.css [hunk 1]
```diff
@@ -317,14 +317,14 @@
...
+  --inputBackgroundColor: var(--inputBackground);
+  --inputBorderColor: var(--surfaceElements);
+  --inputTextColor: var(--primaryElementText);
+	 
+	/* Input Focus */
   --inputFocusBorderColor: var(--primary);
-  /* --inputFocusBoxShadow: 0 0 4px var(--colorSecondaryLighten3); */
-  --inputFocusBoxShadow: inset 0 0 0 1px white,
-    inset 0 0 0 2px var(--colorPrimary);
+  --inputFocusBoxShadow: inset 0 0 0 1px var(--primary),
```
Comment (amalyah) [inline src/css/common/_theme--dark.css:326]: We don't have any shadows in the design files on this component.  @carmenparks Do you want to keep or throw out the Focus Box Shadow for inputs on focus?

**Review by amalyah — CHANGES_REQUESTED** [review]
Pending answer from Carmen

**Comment by im-cr** [comment]
> This appears to be using the wrong search icon for search inputs. Do you have access to the other search icon file?

can we attached the correct icon file to the dev ticket. https://rocketcom.atlassian.net/browse/ASTRO-942

**Comment by amalyah** [comment]
> > This appears to be using the wrong search icon for search inputs. Do you have access to the other search icon file?
> 
> @amalyah can we attached the correct icon file to the dev ticket. https://rocketcom.atlassian.net/browse/ASTRO-942

Attached to ticket

**Review by amalyah — CHANGES_REQUESTED** [review]
Needs search icon attached to dev ticket. Need answer from Visual Design regarding shadow.

File: src/css/common/_theme--dark.css (hunk 1 above)
Comment (carmenparks) [reply src/css/common/_theme--dark.css:326]: With the updated library, we've been removing shadows on components, so let's throw it out here.

Comment (amalyah) [reply src/css/common/_theme--dark.css:326]: @im-cr The verdict from Carmen is throw out component shadows.

File: src/css/common/_theme--light.css [hunk 2]
```diff
@@ -325,11 +325,11 @@
...
-  --inputBackgroundColor: var(--colorWhite);
-  --inputBorderColor: var(--colorQuaternaryDarken1);
-  --inputTextColor: var(--colorBlack);
+  --inputBackgroundColor: var(--inputBackground);
+  --inputBorderColor: var(--primaryLight);
+  --inputTextColor: var(--defaultText);
 
-  --inputFocusBorderColor: var(--colorSecondary);
+  --inputFocusBorderColor: var(--primary);
   --inputFocusBoxShadow: none;
```
Comment (amalyah) [inline src/css/common/_theme--light.css:333]: This one is none, but would it be better to just remove the shadow altogether?

**Review by amalyah — CHANGES_REQUESTED** [review]
Sorry, remove shadows and update search icon and approved by me.

**Review by amalyah — CHANGES_REQUESTED** [review]
-- (accidental extra review) --

File: src/css/common/_theme--light.css (hunk 2 above)
Comment (im-cr) [reply src/css/common/_theme--light.css:330]: yes

**Comment by im-cr** [comment]
@amalyah  can you confirm search icon is correct now. Also, on light theme search icon had a color value of #005a92. This isn't represented in the light theme color values.

**Comment by amalyah** [comment]
> @amalyah can you confirm search icon is correct now. Also, on light theme search icon had a color value of #005a92. This isn't represented in the light theme color values.

Icon is the correct image now. @carmenparks, can you weigh in on the appropriate size of the search icon vs. the clear icon and the spacing between the search icon and the text?

Checking the latest master design files, on light and dark theme, the search icon should be primary-dark.

File: static/css/astro.core.css [hunk 3]
```diff
@@ -2120,7 +2118,9 @@ label {
   padding: 0.5rem 1.875rem 0.5rem 0.5rem;
 }
 
-.rux-form-field input:focus .rux-form-field input:invalid:focus {
+.rux-form-field input:focus,
+.rux-form-field input:invalid:focus,
+.rux-form-field textarea:focus {
   box-shadow: inset 0 0 3px rgb(204, 204, 204);
   box-shadow: inset 0 0 3px var(--colorGray);
```
Comment (amalyah) [inline static/css/astro.core.css:2125]: If it matters, this can also be updated to a newer color variable

File: static/css/astro.core.css [hunk 4]
```diff
@@ -2120,7 +2118,9 @@ label {
   padding: 0.5rem 1.875rem 0.5rem 0.5rem;
 }
 
-.rux-form-field input:focus .rux-form-field input:invalid:focus {
+.rux-form-field input:focus,
+.rux-form-field input:invalid:focus,
+.rux-form-field textarea:focus {
   box-shadow: inset 0 0 3px rgb(204, 204, 204);
```
Comment (amalyah) [inline static/css/astro.core.css:2124]: If this is a normal shadow, I believe that @carmenparks would like it removed.

**Review by amalyah — CHANGES_REQUESTED** [review]

File: static/css/astro.core.css (hunk 4 above)
Comment (im-cr) [reply static/css/astro.core.css:2124]: box-shadows removed

File: static/css/astro.core.css (hunk 3 above)
Comment (im-cr) [reply static/css/astro.core.css]: box shadows removed

**Comment by im-cr** [comment]
@carmenparks can you confirm icon style/visuals

**Comment by carmenparks** [comment]
> > @amalyah can you confirm search icon is correct now. Also, on light theme search icon had a color value of #005a92. This isn't represented in the light theme color values.
> 
> Icon is the correct image now. @carmenparks, can you weigh in on the appropriate size of the search icon vs. the clear icon and the spacing between the search icon and the text?
> 
> Checking the latest master design files, on light and dark theme, the search icon should be primary-dark.

@amalyah I'm not sure if this is what you're asking, but the icons should match how it's depicted in the Sketch library.
Search icon: 15 x 15px
Clear icon: 11 x 11px
Spacing between Search icon and text: 10px

File: src/css/core/input.css [hunk 5]
```diff
@@ -209,8 +211,9 @@
...
 }
 
-.rux-form-field input:focus .rux-form-field input:invalid:focus {
-  box-shadow: inset 0 0 3px var(--colorGray);
+.rux-form-field input:focus,
+.rux-form-field input:invalid:focus,
+.rux-form-field textarea:focus {
   border-color: var(--inputFocusBorderColor);
   outline: none;
   color: black;
```
[the remark this answers is not in the document]
Comment (im-cr) [reply src/css/core/input.css]: text color variable created and update for dark/light

**Review by cskerbo — APPROVED** [review]

**Comment by im-cr** [comment]
@carmenparks those icons are in place. please confirm via the link below:

- https://deploy-preview-42--astro-components.netlify.app/?path=/story/components-form-elements--input-fields

**Comment by carmenparks** [comment]
> @carmenparks those icons are in place. please confirm via the link below:
> 
> * https://deploy-preview-42--astro-components.netlify.app/?path=/story/components-form-elements--input-fields

Looks good @im-cr

**Review by carmenparks — APPROVED** [review]

**Pull request merged**